File: codeforces/ac/0/075/Dgen.cpp
#include <cstdio>
#include <cstdlib>

int main() {
	int n = 50, m = 250000, l = 5000;
	printf("%d %d\n", n, m);
	for (int i = 0; i < n; ++i) {
		printf("%d", l);
		for (int j = 0; j < l; ++j) {
			printf(" %d", rand() % 1998 - 1000);
		}
		puts("");
	}
	for (int i = 0; i < m; ++i) {
		printf("%d%c", 1 + rand() % n, i == m - 1 ? '\n' : ' ');
	}
	return 0;
}

File: codeforces/ac/1/173/Dgen.rb
n = 100000
m = 100000
e = Array.new(m) {
  [rand(n) / 2 * 2, rand(n) / 2 * 2 + 1]
}
puts "#{n} #{m}"
puts e.map{|i| i.map(&:succ) * ' '}
